dotCMS/src/main/java/com/dotcms/rest/api/v1/index/ESIndexResource.java

```diff
@@ -67,392 +67,351 @@
 @Path("/v1/esindex")
 public class ESIndexResource {
 
+    private enum IndexAction{
+        ACTIVATE,
+        DEACTIVATE,
+        CLEAR,
+        OPEN,
+        CLOSE;
+        
+        public static IndexAction fromString(String actionStr) {
+            for(IndexAction action: IndexAction.values()) {
+                if(action.name().equalsIgnoreCase(actionStr)) {
+                    return action;
+                }
+                
+            }
+            return ACTIVATE;
+        }
+    }
+    
+    
+    
+    
+    
+    
+    
 	private final ESIndexAPI indexAPI;
 	private final ESIndexHelper indexHelper;
-	private final ResponseUtil responseUtil;
-	private final WebResource webResource;
-	private final LayoutAPI layoutAPI;
-	private final IndiciesAPI indiciesAPI;
+    private final ContentletIndexAPI idxApi;
+
 
 	public ESIndexResource(){
 		this.indexAPI = APILocator.getESIndexAPI();
 		this.indexHelper = ESIndexHelper.getInstance();
-		this.responseUtil = ResponseUtil.INSTANCE;
-		this.webResource = new WebResource();
-		this.layoutAPI = APILocator.getLayoutAPI();
-		this.indiciesAPI = APILocator.getIndiciesAPI();
+	    this.idxApi = APILocator.getContentletIndexAPI();
+
+
 	}
 
 	@VisibleForTesting
-	ESIndexResource(ESIndexAPI indexAPI, ESIndexHelper indexHelper, ResponseUtil responseUtil,
+	ESIndexResource(ESIndexAPI indexAPI, ESIndexHelper indexHelper,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'indexAPI' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)